core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -164,11 +167,25 @@ long memorySize() {
 
         updateIndexAndNormalize(engaged, newChunks);
 
-        for (Chunk chunk : engaged
-        ) {
+        for (Chunk chunk : engaged) {
+            ConcurrentSkipListSet<Iter> viewingIterators = chunk.getSignedIterators();
```

Comment:
add a comment explaining what is the goal of this checks - this is a crucial piece of code
also it helps reviewing